cli/parse.go

```diff
@@ -510,6 +510,15 @@ func appendFile(fpath string, argDef *cmds.Argument, recursive, hidden bool) (fi
 		if !recursive {
 			return nil, fmt.Errorf(notRecursiveFmtStr, fpath, cmds.RecShort)
 		}
+	} else if (stat.Mode() & os.ModeNamedPipe) != 0 {
+		// Special case pipes that are provided directly on the command line
+		// We do this here instead of go-ipfs-files, as we need to differentiate between
+		// recursive(unsupported) and direct(supported) mode
+		file, err := os.Open(fpath)
+		if err != nil {
+			return nil, err
+		}
+		return files.NewReaderPathFile(fpath, file, stat)
```

Comment:
The path & stat should be nil.

* The path should only refer to files that can _actually_ be read. The filestore/urlstore features will use it.
* The stat will be used to determine the size of the file. It'll give us _very_ incorrect information in this case.

Reply:
`path` is a string, so I went with `""` ( thought of using `newMessageReader` but proably incorrect in this case ). There were zero changes needed in the go-ipfs test suite ( aside from https://github.com/ipfs/go-ipfs/pull/6998 obviously ), so this seems like the right thing to do.